mlir/lib/Dialect/MQTOpt/Transforms/ToQuantumComputationPattern.cpp

```diff
@@ -76,12 +79,17 @@ struct ToQuantumComputationPattern final : mlir::OpRewritePattern<AllocOp> {
     }
     for (size_t i = 0; i < currentQubitVariables.size(); i++) {
       size_t qubitArrayIndex = 0;
-      if (auto opResult =
-              llvm::dyn_cast<mlir::OpResult>(currentQubitVariables[i])) {
-        qubitArrayIndex = opResult.getResultNumber();
+      if (currentQubitVariables[i] != nullptr) {
+        auto opResult =
+            llvm::dyn_cast<mlir::OpResult>(currentQubitVariables[i]);
+        qubitArrayIndex =
+            opResult
+                .getResultNumber(); // e.g. is 1 if it comes from an extract op
```

Comment:
The formatting here irritates me
```suggestion
        // e.g. is 1 if it comes from an extract op
        qubitArrayIndex = opResult.getResultNumber(); 
```